### ISSUE TITLE
Bug in `line_project_pointC2` when the line is almost horizontal

### ISSUE THREAD

**Issue by cjamin** [CGAL/cgal#3301]
Using kernel `Simple_cartesian<double>`.

I have a `Line_2`:
a = -4.2885603045067812e-18
b = 1
c = 250.73609999999996

When I try to project the point (35.306000000000004, 250.69800000000001) onto the line, I get the following projected point : (0, 250.73609999999996).

**Comment by cjamin** [comment]
I also tried this code, but there must be something wrong in it since I got much more wrong results. I did not really look into it, I just guessed it had not been really tested.

**Comment by mglisse** [comment]
Previously mentioned in https://sympa.inria.fr/sympa/arc/cgal-develop/2015-05/msg00041.html (but I don't think there is much useful information there).

**Comment by lrineau** [comment]
[**@cjamin**](https://github.com/cjamin) commented on [Sep 6, 2018, 6:00 PM GMT+2](https://github.com/CGAL/cgal/issues/3301#issuecomment-419149008 "2018-09-06T16:00:38Z - Replied by Github Reply Comments"):
> For now, I'm using a workaround in my code which widens a little bit the `a,b=0` tests: something like `if (a > -1e-8 && a < 1e-8)`.

I think the test should compare the absolute values of `la` and `lb` and decide from them if the line is more vertical that horizontal.

**Comment by cjamin** [comment]
Yes, it is true that both `la` and `lb` can be very small. In practice, I'm not sure this can happen except when using the first constructor, but I'll look into it. Thank you.

**Comment by afabri** [comment]
Hello @cjamin, I think I improved the situation.

**Comment by cjamin** [comment]
Hi @afabri,
Thank you very much! I'll look into it as soon as possible.

**Comment by afabri** [comment]
Why enhancement? Isn't this just  a bug fix and should go in 4.13.1 ?

**Comment by lrineau** [comment]
I agree. I changed that. It could even have been a bug fix for CGAL-4.12.2, if the branch was not created from `master`.

**Comment by psarahdactyl** [comment]
So will this fix be in 4.12.2 or 4.13.1?

**Comment by lrineau** [comment]
It will for sure be in CGAL-4.13.1, but for the moment it is not planed in 4.12.2, unless @afabri rebases the branch of the PR.

**Comment by lrineau** [comment]
I have rebased the branch, and tested the testsuite of Kernel_23. The fix will go into CGAL-4.12.2 as well.
